File: Data Structures & Algorithms/two-integer-sum/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const smap = {}
        for(let i = 0; i < nums.length; i++){
            if(target-nums[i] in smap){
                return [smap[target-nums[i]],i]
            }else{
                smap[nums[i]] = i
            }
        }
    }
}

File: Data Structures & Algorithms/jump-game/submission-7.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        def cantJump(i):
            if i >= len(nums)-1:
                return True
            for j in range(1,nums[i]+1):
                if cantJump(i+j):
                    return True
            
        return True if cantJump(0) else False

